### PR TITLE
planner: Fix IndexJoin's inner TableScan's Range when explain/desc

### LINKED ISSUE #14822: planner: IndexJoin's inner TableScan's range should not be decided by columns in `OtherCondition` when EXPLAIN
## Feature Request

**Is your feature request related to a problem? Please describe:**

```
mysql> explain select /*+ TIDB_INLJ(t2)*/ * from t1, t2 where t1.a = t2.a and t1.b = t2.b;
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
| id                       | count | task      | operator info                                                                                                                  |
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
| IndexJoin_10             | 1.25  | root      | inner join, inner:TableReader_9, outer key:jointest.t1.a, inner key:jointest.t2.a, other cond:eq(jointest.t1.b, jointest.t2.b) |
| ├─TableReader_20         | 1.00  | root      | data:Selection_19                                                                                                              |
| │ └─Selection_19         | 1.00  | cop[tikv] | not(isnull(jointest.t1.a)), not(isnull(jointest.t1.b))                                                                         |
| │   └─TableScan_18       | 1.00  | cop[tikv] | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo                                                                    |
| └─TableReader_9          | 1.00  | root      | data:Selection_8                                                                                                               |
|   └─Selection_8          | 1.00  | cop[tikv] | not(isnull(jointest.t2.b))                                                                                                     |
|     └─TableScan_7        | 1.00  | cop[tikv] | table:t2, range: decided by [jointest.t1.a jointest.t1.b], keep order:false, stats:pseudo                                      |
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
7 rows in set (0.01 sec)
```

**Describe the feature you'd like:**

In the query above, the inner side `TableScan` is only affected by `t1.a`. But the `EXPLAIN` shows that it is decided by both `t1.a` and `t1.b`. So I suggest the explain result should be:
```
mysql> explain select /*+ TIDB_INLJ(t2)*/ * from t1, t2 where t1.a = t2.a and t1.b = t2.b;
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
| id                       | count | task      | operator info                                                                                                                  |
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
| IndexJoin_10             | 1.25  | root      | inner join, inner:TableReader_9, outer key:jointest.t1.a, inner key:jointest.t2.a, other cond:eq(jointest.t1.b, jointest.t2.b) |
| ├─TableReader_20         | 1.00  | root      | data:Selection_19                                                                                                              |
| │ └─Selection_19         | 1.00  | cop[tikv] | not(isnull(jointest.t1.a)), not(isnull(jointest.t1.b))                                                                         |
| │   └─TableScan_18       | 1.00  | cop[tikv] | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo                                                                    |
| └─TableReader_9          | 1.00  | root      | data:Selection_8                                                                                                               |
|   └─Selection_8          | 1.00  | cop[tikv] | not(isnull(jointest.t2.b))                                                                                                     |
|     └─TableScan_7        | 1.00  | cop[tikv] | table:t2, range: decided by [jointest.t1.a], keep order:false, stats:pseudo                                      |
+--------------------------+-------+-----------+--------------------------------------------------------------------------------------------------------------------------------+
7 rows in set (0.01 sec)
```

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by gauss1314**
### What problem does this PR solve?

This fix #14822 
IndexJoin's inner TableScan's range should only be decided by columns when conditions contain inner primary key  for explain/desc

### Check List 

Tests 
 - Unit test
 - Integration test

### CODE REVIEW

File: planner/core/testdata/integration_suite_in.json [hunk 1]
```diff
@@ -80,5 +80,11 @@
       "desc select /*+ read_from_storage(tikv[t]) */ avg(a) from t",
       "desc select /*+ read_from_storage(tiflash[t]) */ avg(a) from t"
     ]
+  },
+  {
+    "name": "TestIndexJoinTableRange",
+    "cases": [
+      "desc select /*+ TIDB_INLJ(t2)*/ * from t1, t2 where t1.a = t2.a and t1.b = t2.b"
```
Comment (francis0407) [inline planner/core/testdata/integration_suite_in.json]: Also add this case:
```
desc select /*+ TIDB_INLJ(t2）*/ * from t1, t2 where t1.a = t2.a and t1.b = t2.a and t1.b = t2.b
```

File: planner/core/exhaust_physical_plans.go [hunk 2]
```diff
@@ -556,6 +556,8 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 		}
 		pkMatched = true
 		keyOff2IdxOff[i] = 0
+		// There is only one EqualCondition for IndexJoin to construct Range when it is a primary key. For issue #14822.
```
Comment (francis0407) [inline planner/core/exhaust_physical_plans.go]: See the comment I left before. The range could be decided by more than one EqualCondition, only if these conditions contain inner primary key. For example: `select * from t1, t2 where t1.a = t2.pk and t1.b = t2.pk`, in this query, both `t1.a` and `t1.b` shoule be in `outerJoinKeys`.

Comment (gauss1314) [reply planner/core/exhaust_physical_plans.go]: Sorry for the misread. I have change it now.

File: planner/core/exhaust_physical_plans.go [hunk 3]
```diff
@@ -548,15 +548,19 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 		return nil
 	}
 	keyOff2IdxOff := make([]int, len(innerJoinKeys))
+	newOuterJoinKeys := make([]*expression.Column, 0)
 	pkMatched := false
 	for i, key := range innerJoinKeys {
-		if !key.Equal(nil, pkCol) {
+		if !key.Equal(nil, pkCol) && !outerJoinKeys[i].Equal(nil, pkCol) {
```
Comment (francis0407) [inline planner/core/exhaust_physical_plans.go]: We don't need to change this line. `pkCol` is from innerTable, it won't be equal to any of the outerJoinKey.

**Pull request merged**